crepecake/src/main/java/com/google/cloud/tools/crepecake/tar/TarStreamBuilder.java

```diff
@@ -44,22 +43,17 @@ private static void writeEntriesAsTarArchive(
         new TarArchiveOutputStream(tarByteStream)) {
       for (TarArchiveEntry entry : entries) {
         tarArchiveOutputStream.putArchiveEntry(entry);
-        InputStream contentStream = new BufferedInputStream(new FileInputStream(entry.getFile()));
-        ByteStreams.copy(contentStream, tarArchiveOutputStream);
+        if (entry.isFile()) {
```

Comment:
Are directories just created as needed when adding files?

Reply:
Here, directories are still added as `putArchiveEntry` puts the directory into the archive stream. Directories don't have content so that's why this code is only for regular files.